core/src/transport.rs

```diff
@@ -229,82 +228,39 @@ pub trait Transport {
     }
 }
 
-/// The ID of a single listener.
-#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub struct ListenerId {
-    id: u64,
-    transport_id: TypeId,
-}
-
-impl ListenerId {
-    /// Creates a new `ListenerId`.
-    pub fn new<T: Any>(id: u64) -> Self {
-        Self {
-            id,
-            transport_id: TypeId::of::<T>(),
-        }
-    }
-
-    /// Returns the next id
-    pub fn next_id(&mut self) -> Self {
-        let current = *self;
-        self.id += 1;
-        current
-    }
-
-    pub fn map_type<T: Any>(mut self) -> Self {
-        self.transport_id = TypeId::of::<T>();
-        self
-    }
-}
-
 /// Event produced by [`Transport`]s.
 ///
 /// Transports are expected to produce `Incoming` events only for
 /// listen addresses which have previously been announced via
 /// a `NewAddress` event and which have not been invalidated by
 /// an `AddressExpired` event yet.
+#[derive(Clone, PartialEq)]
 pub enum TransportEvent<TUpgr, TErr> {
     /// A new address is being listened on.
-    NewAddress {
-        /// The listener that is listening on the new address.
-        listener_id: ListenerId,
-        /// The new address that is being listened on.
-        listen_addr: Multiaddr,
-    },
+    NewAddress(Multiaddr),
```

Comment:
Not moving from a struct to a tuple variant would reduce the diff a bit I think.